app/moves/middleware/set-body.outgoing.js

```diff
@@ -0,0 +1,19 @@
+const { get, set } = require('lodash')
+
+function setBodyOutgoing(req, res, next) {
+  const { dateRange, locationId } = req.params
```

Comment:
This is quite literally the same as `set incoming`. Can you incorporate them together, maybe through a function that accepts the type of move as parameters and returns the middleware function?

Reply:
You're right. I did toy with that idea too. The two major differences are with the property that is used to set it on the body (which is currently to aid re-using multiple of these middleware on the dashboard and I wonder if there's a better way to actually do that too) and the other is that one uses `fromLocationId` and the other `toLocationId`. 

Both could be passed into a function which returns the middleware but I didn't know if that would add too much logic and make it harder to read. What do you think?

It would look something like this:

```javascript
// middleware
function setBodyMoves(bodyKey, locationKey) {
  return function handleBody(req, res, next) {
    const { dateRange, locationId } = req.params
    const locations =
      locationId ||
      get(req.session, 'user.locations', [])
        .map(location => location.id)
        .join(',')

    set(req, `body.${bodyKey}`, {
      dateRange,
      [locationKey]: locations,
    })

    next()
  }
}


// moves/index.js
setBodyMoves('outgoing', 'fromLocationId')
...
setBodyMoves('incoming', 'toLocationId')
```

If that still feels okay to read I'll update them to use this shared middleware.

Reply:
Seems good to me

Reply:
@barbaracassani if you're happy too I'll do that as a fixup